common/softwarecontainer-common.cpp

```diff
@@ -110,6 +111,16 @@ std::string parentPath(const std::string &path_)
     return parentPath;
 }
 
+std::string buildPath(const std::string &arg1, const std::string &arg2)
+{
+    return Glib::build_filename(arg1, arg2);
+}
+
+std::string buildPath(const std::string &arg1, const std::string &arg2, const std::string &arg3)
+{
+    return Glib::build_filename(arg1, arg2, arg3);
```

Comment:
In order to lessen the work needed to swap Glib for some other lib in the future, it would be really neat if this was defined as a nested case of the one with two arguments.